core/input/input.cpp

```diff
@@ -475,7 +475,7 @@ void Input::joy_connection_changed(int p_idx, bool p_connected, String p_name, S
 	}
 	joy_names[p_idx] = js;
 
-	emit_signal(SNAME("joy_connection_changed"), p_idx, p_connected);
+	call_deferred("emit_signal", SNAME("joy_connection_changed"), p_idx, p_connected);
```

Comment:
I'd add a comment to clarify, like "Defer to emit on the main thread." (IIUC).

Reply:
BTW I'm not good with multithreading, but would deferring the call to `joy_connection_changed` in the Linux input code solve the issue similarly?

If only some platforms do this in a thread, maybe they should be responsible for deferring to the main thread where needed?

Though this one claims to be thread-safe so I guess this helps provide this guarantee?